src/Bicep.LangServer/Handlers/InsertResourceHandler.cs

```diff
@@ -279,9 +279,15 @@ private static SyntaxBase ConvertJsonElement(JsonElement element)
                 case JsonValueKind.String:
                     return SyntaxFactory.CreateStringLiteral(element.GetString()!);
                 case JsonValueKind.Number:
-                    if (element.TryGetInt32(out var intValue))
+                    if (element.TryGetInt64(out var intValue))
                     {
-                        return SyntaxFactory.CreateIntegerLiteral(element.GetInt32()!);
+                        if (intValue >= 0)
+                        {
+                            return SyntaxFactory.CreateIntegerLiteral((ulong)intValue);
+                        } else
```

Comment:
[nit] Format: place `else` on new line. Or we could replace `if-else` with the conditional operator `?:` since it's a simple condition.